packages/shared/src/modules/routes/libs/validation-schemas/route-search-query.validation-schema.ts

```diff
@@ -5,7 +5,28 @@ import {
 	RoutesValidationRule,
 } from "../enums/enums.js";
 
+const categoryKey = z
+	.string()
+	.regex(/^[a-zA-Z-]+$/, {
+		message: RoutesValidationMessage.CATEGORY_INVALID_KEY,
+	})
+	.min(RoutesValidationRule.CATEGORY_MINIMUM_LENGTH, {
+		message: RoutesValidationMessage.CATEGORY_MINIMUM_LENGTH,
+	})
+	.max(RoutesValidationRule.CATEGORY_MAXIMUM_LENGTH, {
+		message: RoutesValidationMessage.CATEGORY_MAXIMUM_LENGTH,
+	});
+
 const routesSearchQuery = z.object({
+	categories: z
+		.preprocess((value) => {
+			if (value === undefined || value === null) {
+				return;
+			}
+
+			return Array.isArray(value) ? (value as string[]) : [value];
+		}, z.array(categoryKey))
+		.optional(),
```

Comment:
take a look at how processing is done for pointsOfInterestSearchQuery. let's transform the value here and remove the same transformation inside the controller
```suggestion
			.transform(parseToFloat)
			.pipe(
```